test/integration/prometheus/uid_verification_test.go

```diff
@@ -0,0 +1,255 @@
+			"kube_node_status_capacity_cpu_cores", "kube_node_status_allocatable",
+			"kube_node_status_allocatable_cpu_cores", "kube_node_status_allocatable_memory_bytes",
+			"kube_node_labels", "kube_node_status_condition":
+			if item.Metric.Node != "" {
+				resourceName = item.Metric.Node // cluster-scoped
+			}
+		// Pod metrics
+		case "kube_pod_labels", "kube_pod_owner", "kube_pod_container_status_running",
+			"kube_pod_container_status_terminated_reason", "kube_pod_container_status_restarts_total",
+			"kube_pod_container_resource_requests", "kube_pod_container_resource_limits",
+			"kube_pod_container_resource_limits_cpu_cores", "kube_pod_container_resource_limits_memory_bytes",
+			"kube_pod_status_phase":
+			if item.Metric.Pod != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.Pod) // namespace-scoped
+			}
+		// PVC metrics
+		case "kube_persistentvolumeclaim_resource_requests_storage_bytes",
+			"kube_persistentvolumeclaim_info":
+			if item.Metric.PersistentVolumeClaim != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.PersistentVolumeClaim) // namespace-scoped
+			}
+		// PV metrics
+		case "kube_persistentvolume_capacity_bytes", "kube_persistentvolume_status_phase",
+			"kubecost_pv_info":
+			if item.Metric.PersistentVolume != "" {
+				resourceName = item.Metric.PersistentVolume // cluster-scoped
+			}
+		// Service metrics
+		case "service_selector_labels":
+			if item.Metric.Service != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.Service) // namespace-scoped
+			}
+		// StatefulSet metrics
+		case "statefulSet_match_labels":
+			if item.Metric.StatefulSet != "" && item.Metric.Namespace != "" {
+				resourceName = fmt.Sprintf("%s/%s", item.Metric.Namespace, item.Metric.StatefulSet) // namespace-scoped
+			}
+		}
+
+		// Count all resources found
+		if resourceName != "" {
+			totalResources++
+			// Only add to uidMap if UID is present
+			if uid != "" {
+				uidMap[resourceName] = uid
+			}
+		}
+	}
+
+	return uidMap, totalResources
+}
+
+// TestPodUIDVerification tests that pods have valid UIDs in its metrics
+func TestPodUIDVerification(t *testing.T) {
+	podMetrics := []string{"kube_pod_labels", "kube_pod_owner", "kube_pod_container_status_running",
+		"kube_pod_container_status_terminated_reason", "kube_pod_container_status_restarts_total",
+		"kube_pod_container_resource_requests", "kube_pod_container_resource_limits",
+		"kube_pod_container_resource_limits_cpu_cores", "kube_pod_container_resource_limits_memory_bytes",
+		"kube_pod_status_phase"}
+	testResourceUIDVerification(t, "pods", podMetrics)
+}
+
+// TestDeploymentUIDVerification tests that deployments have valid UIDs in its metrics
+func TestDeploymentUIDVerification(t *testing.T) {
+	deploymentMetrics := []string{"deployment_match_labels", "kube_deployment_spec_replicas",
```

Comment:
I would re-write all these as test cases, then re-use stuff like the prom client

Reply:
+1